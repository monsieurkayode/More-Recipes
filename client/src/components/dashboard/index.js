/* jshint esversion: 6 */
import UserProfile from './UserProfile';
import UserRecipe from './UserRecipe';
import WelcomeDisplay from './WelcomeDisplay';
import DashboardPanel from './DashboardPanel';
import SideNavDashboard from './SideNavDashboard';
import UserFavoriteRecipe from './UserFavoriteRecipe';
import UserNotification from './UserNotification';

export {
  UserProfile,
  UserRecipe,
  UserFavoriteRecipe,
  UserNotification,
  WelcomeDisplay,
  SideNavDashboard,
  DashboardPanel
};
