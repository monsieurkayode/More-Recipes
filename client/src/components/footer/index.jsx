import React from 'react';

/**
 * Footer
 * @function Footer
 *
 * @param {void} void
 *
 * @return {JSX} JSX
 */
const Footer = () => (
  <footer className="page-footer blue-grey darken-2">
    <div className="row">
      <div className="col l3 social-icon-container">
        <ul>
          <li>
            <div>
              <span>
                <i className="fa fa-2x fa-github teal-text" />
              </span>
            </div>
          </li>
          <li>
            <div>
              <span>
                <i className="fa fa-2x fa-facebook teal-text" />
              </span>
            </div>
          </li>
          <li>
            <div>
              <span>
                <i className="fa fa-2x fa-twitter teal-text" />
              </span>
            </div>
          </li>
          <li>
            <div>
              <span>
                <i className="fa fa-2x fa-google teal-text" />
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div className="footer-copyright">
      <div className="container">
        Â© Andela 2018
        <a
          className="grey-text text-lighten-4 right"
          href="/"
        >
        Created by Kayode Adeola
        </a>
      </div>
    </div>
  </footer>
);

export default Footer;
