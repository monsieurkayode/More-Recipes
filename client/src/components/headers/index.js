/* jshint esversion: 6 */
import Banner from './Banner';
import DashboardNavbar from './DashboardNavbar';
import HomeNavbar from './HomeNavbar';
import LandingNavbar from './LandingNavbar';
import UserMenu from './UserMenu';
import SearchBar from './SearchBar';

export {
  Banner,
  DashboardNavbar,
  HomeNavbar,
  LandingNavbar,
  UserMenu,
  SearchBar
};
