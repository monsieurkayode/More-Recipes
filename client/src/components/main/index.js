/* jshint esversion: 6 */
import Category from './Category';
import Recipes from './Recipes';
import RecipeItem from './RecipeItem';
import TopRecipeItem from './TopRecipeItem';
import TopRecipes from './TopRecipes';
import SignupForm from './SignupForm';
import SigninForm from './SigninForm';
import TextField from './TextField';
import PostRecipe from './PostRecipe';
import EditRecipe from './EditRecipe';
import SideNav from './SideNav';
import WelcomeMessage from './WelcomeMessage';
import Loader from './Loader';

export {
  Category,
  Recipes,
  RecipeItem,
  TopRecipeItem,
  TopRecipes,
  PostRecipe,
  EditRecipe,
  SignupForm,
  SigninForm,
  TextField,
  SideNav,
  WelcomeMessage,
  Loader
};
