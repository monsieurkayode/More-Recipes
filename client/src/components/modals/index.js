/* jshint esversion: 6 */
import DeleteModal from './DeleteModal';

export default DeleteModal;
