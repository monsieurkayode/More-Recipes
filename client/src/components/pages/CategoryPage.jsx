import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import isEmpty from 'lodash/isEmpty';
import ReactPaginate from 'react-paginate';
import classnames from 'classnames';
import PropTypes from 'proptypes';

import {
  upvoteAction,
  downvoteAction,
  fetchRecipesByCategory,
} from '../../actions';
import { HomeNavbar, Banner } from '../headers';
import { RecipeItem, Loader, SideNav } from '../main';
import Footer from '../footer';
import materializeJavascript from '../../utils/materializeJavascript';
import categories from '../../../../shared/categories';
import pascalCase from '../../utils/pascalCase';
import resetPage from '../../utils/resetPage';
import sad from '../../../assets/css/img/sad.png';
import happySmiley from '../../../assets/css/img/happy_smiley.png';

/**
 * @summary - CategoryPage class declaration
 * @class CategoryPage
 * @extends {Component}
 */
export class CategoryPage extends Component {
  /**
   * Component constructor
   * @param {object} props
   * @memberOf IndexPage
   */
  constructor(props) {
    super(props);
    this.state = {
      isLoading: true,
      firstLoad: true,
      limit: 9,
      changeSmiley: false
    };
    this.toggleSmiley = this.toggleSmiley.bind(this);
    this.setPagination = this.setPagination.bind(this);
    this.hasRecipes = this.hasRecipes.bind(this);
    this.handlePageClick = this.handlePageClick.bind(this);
  }

  /**
   * @method componentDidMount
   *
   * @returns {undefined}
   */
  componentDidMount() {
    const { categoryName } = this.props.match.params;
    const { limit } = this.state;
    const currentPage = localStorage.getItem('currentCategoryPage');
    this.props.fetchRecipesByCategory(currentPage, limit, categoryName)
      .then(() => this.setState({ firstLoad: false, isLoading: false }));
  }

  /**
   * @method componentDidUpdate
   *
   * @returns {undefined}
   */
  componentDidUpdate() {
    materializeJavascript();
  }

  onClick = () => {
    resetPage();
  }

  /**
   * @method setPagination
   *
   * @param {void} void
   *
   * @returns {object} pagination props
   */
  setPagination() {
    if (this.hasRecipes()) {
      const { pagination: { page, pageCount } } = this.props.recipes;
      return {
        page,
        pageCount
      };
    }
    return {
      page: 1,
      pageCount: 1
    };
  }

  /**
   * Toggle smiley from sad face to
   * happy face
   * @method toggleSmiley
   *
   * @returns {undefined}
   */
  toggleSmiley() {
    const { changeSmiley } = this.state;
    this.setState({ changeSmiley: !changeSmiley });
  }

  /**
   * Checks if there are recipes to display
   * @method hasRecipes
   *
   * @returns {boolean} boolean
   */
  hasRecipes() {
    if (isEmpty(this.props.recipes)) {
      return false;
    }
    return !isEmpty(this.props.recipes.recipes);
  }

  /**
  * Handle page selection for pagination
  * @method handlePageClick
  *
  * @param {object} data
  *
  * @returns {void}
  */
  handlePageClick({ selected }) {
    this.setState({ isLoading: true });
    const page = selected + 1;
    const { limit } = this.state;
    localStorage.setItem('currentCategoryPage', page);
    const currentPage = localStorage.getItem('currentCategoryPage');
    const { categoryName } = this.props.match.params;
    this.props.fetchRecipesByCategory(currentPage, limit, categoryName)
      .then(() => this.setState({ isLoading: false }));
  }

  /**
  * Render recipes by category
  * @method renderCategoryRecipes
  *
  * @param {number} index
  *
  * @returns {JSX} JSX
  */
  renderCategoryRecipes = (index) => {
    const recipe = this.props.recipes.recipes[index];
    return (
      <RecipeItem
        grid="l4 m6"
        key={recipe.id}
        index={recipe.id}
        recipe={recipe}
        upvote={this.props.upvoteAction}
        downvote={this.props.downvoteAction}
      />
    );
  }

  /**
  * Render category link items
  * @method renderCategoryRecipes
  *
  * @param {number} index
  *
  * @returns {JSX} JSX
  */
  renderCategoryLinks = (index) => {
    const category = categories[index];
    const { categoryName } = this.props.match.params;
    const linkClass = classnames({ active: category === categoryName });
    return (
      <li key={category}>
        <Link
          to={`/category/${category}`}
          className={`collapsible-header truncate ${linkClass}`}
          onClick={this.onClick}
        >
          {pascalCase(category)}
        </Link>
      </li>
    );
  }

  /**
   * @method renderPagination
   *
   * @param {void} void
   *
   * @returns {JSX} JSX
   */
  renderPagination = () => {
    const { page, pageCount } = this.setPagination();
    return (
      <div className="pagination-container">
        <ReactPaginate
          previousLabel={<i className="fa fa-chevron-left" />}
          nextLabel={<i className="fa fa-chevron-right" />}
          breakLabel={<a href="">...</a>}
          breakClassName={'break-me'}
          pageCount={pageCount}
          initialPage={page - 1}
          marginPagesDisplayed={2}
          pageRangeDisplayed={5}
          onPageChange={this.handlePageClick}
          disableInitialCallback
          containerClassName={'pagination'}
          subContainerClassName={'pages pagination'}
          activeClassName={'active'}
        />
      </div>
    );
  }

  /**
   * Renders the component
   * @method render
   *
   * @returns {JSX} JSX
   */
  render() {
    const { recipes } = this.props.recipes;
    const { isLoading, firstLoad } = this.state;
    const { categoryName } = this.props.match.params;
    return (
      <div id="category-page-container">
        { firstLoad ?
          <Loader /> :
          <div>
            <HomeNavbar />
            <Banner />
            <div>
              <h4 className="center-align category-header">
                {pascalCase(categoryName)}
              </h4>
              <div className="row">
                <div className="col l2 offset-l1 m2 hide-on-small-only">
                  <span className="category-section-header">
                    <b>Categories</b>
                  </span>
                  <p className="divider" />
                  <div className="category">
                    <ul className="collapsible z-depth-0">
                      {categories
                        .map(
                          (category, index) => this.renderCategoryLinks(index)
                        )
                      }
                    </ul>
                  </div>
                </div>
                <div className="col l8 m10" >
                  <span
                    className="recipe-section-header"
                  >
                    <b>Recipes</b>
                  </span>
                  <p className="divider" />
                  <div className="row">
                    {!isLoading ?
                      Object
                        .keys(recipes)
                        .sort((a, b) => b - a)
                        .map(index => this.renderCategoryRecipes(index))
                      :
                      <Loader />
                    }
                  </div>
                  {!this.hasRecipes() &&
                  <div className="center-align">
                    <img
                      id="smiley"
                      src={this.state.changeSmiley ?
                        happySmiley : sad}
                      alt=""
                    />
                    <h6>
                      No recipe has been created for {pascalCase(categoryName)}
                      . Care to <Link
                        onMouseEnter={this.toggleSmiley}
                        onMouseLeave={this.toggleSmiley}
                        to="/recipes/new"
                      >add</Link> one?
                    </h6>
                  </div>
                  }
                  {this.hasRecipes() && this.renderPagination()}
                </div>
              </div>
            </div>
            <SideNav />
            <Footer />
          </div>
        }
      </div>
    );
  }
}

const mapStateToProps = ({ recipes }) => ({
  pagination: recipes.pagination,
  recipes
});

CategoryPage.defaultProps = {
  recipes: {}
};

CategoryPage.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      categoryName: PropTypes.string.isRequired
    })
  }).isRequired,
  fetchRecipesByCategory: PropTypes.func.isRequired,
  upvoteAction: PropTypes.func.isRequired,
  downvoteAction: PropTypes.func.isRequired,
  recipes: PropTypes.shape({
    recipes: PropTypes.shape({}),
    pagination: PropTypes.shape({})
  })
};

export default connect(mapStateToProps,
  {
    upvoteAction,
    downvoteAction,
    fetchRecipesByCategory
  }
)(CategoryPage);
