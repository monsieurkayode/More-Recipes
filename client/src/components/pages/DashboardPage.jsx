import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'proptypes';

import {
  logoutAction,
  fetchUserRecipes,
  fetchUserFavorites,
  deletePost,
  removeFavorite,
  selectRecipe,
  isFetching
} from '../../actions';
import { DashboardNavbar } from '../headers';
import {
  DashboardPanel,
  SideNavDashboard,
  UserRecipe,
  UserFavoriteRecipe,
  UserProfile,
  UserNotification
} from '../dashboard';
import DeleteModal from '../modals';
import { Loader } from '../main';
import materializeJavascript from '../../utils/materializeJavascript';

/**
 * @summary - DashboardPage class declaration
 * @class DashboardPage
 * @extends {Component}
 */
export class DashboardPage extends Component {
  /**
   * @method componentDidMount
   *
   * @returns {undefined}
   */
  componentDidMount() {
    const currentPage = localStorage.getItem('currentPageUserRecipes');
    const currentPageFav = localStorage.getItem('currentPageUserFavorites');
    this.props.isFetching(true, 'Dashboard');
    this.props.fetchUserRecipes(currentPage);
    this.props.fetchUserFavorites(currentPageFav);
  }

  /**
   * @method componentDidUpdate
   *
   * @param {void} void
   *
   * @returns {void}
   */
  componentDidUpdate() {
    materializeJavascript();
  }

  /**
   * Renders the component
   * @method render
   *
   * @returns {JSX} JSX
   */
  render() {
    const { isLoading } = this.props;
    const deletePostDialog = 'Delete recipe';
    const removeFavDialog = 'Remove from favorite';
    const { route } = this.props.match.params;
    return (
      <div>
        {isLoading ?
          <Loader /> :
          <div>
            <DashboardNavbar {...this.props} />
            <div className="row">
              <DashboardPanel user={this.props.user} />
              {
                route === 'profile' &&
                <UserProfile selected={route} {...this.props} />
              }
              {
                route === 'recipes' &&
                <UserRecipe selected={route} {...this.props} />
              }
              {
                route === 'favorites' &&
                <UserFavoriteRecipe selected={route} {...this.props} />
              }
              {
                route === 'notifications' &&
                <UserNotification selected={route} {...this.props} />
              }
            </div>
            <DeleteModal
              selected={route}
              id={this.props.selectedRecipe}
              dialog={route === 'recipes' ? deletePostDialog : removeFavDialog}
              handleAction={route === 'recipes' ?
                this.props.deletePost :
                this.props.removeFavorite
              }
            />
            <SideNavDashboard user={this.props.user} />
          </div>}
      </div>
    );
  }
}

const mapStateToProps = ({
  signinState,
  userRecipes,
  userFavorites,
  selectedRecipe,
  isLoading
}) => ({
  user: signinState.user,
  isAuthenticated: signinState.isAuthenticated,
  userRecipes,
  userFavorites,
  selectedRecipe,
  isLoading: isLoading.dashBoardIsLoading,
  isLoadingRecipes: isLoading.userRecipesIsLoading,
  isLoadingFavorites: isLoading.userFavoritesIsLoading
});

DashboardPage.defaultProps = {
  user: {},
  selectedRecipe: 0,
  userFavorites: {},
  userRecipes: {},
};

DashboardPage.propTypes = {
  user: PropTypes.shape({}),
  fetchUserRecipes: PropTypes.func.isRequired,
  userRecipes: PropTypes.shape({}),
  userFavorites: PropTypes.shape({}),
  fetchUserFavorites: PropTypes.func.isRequired,
  selectedRecipe: PropTypes.number,
  deletePost: PropTypes.func.isRequired,
  removeFavorite: PropTypes.func.isRequired,
  isFetching: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      route: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

export default connect(mapStateToProps,
  {
    logoutAction,
    fetchUserRecipes,
    fetchUserFavorites,
    deletePost,
    removeFavorite,
    selectRecipe,
    isFetching
  }
)(DashboardPage);
