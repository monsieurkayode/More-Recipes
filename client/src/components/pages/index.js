/* jshint esversion: 6 */
import SigninPage from './SigninPage';
import SignupPage from './SignupPage';
import DashboardPage from './DashboardPage';
import NotFoundPage from './NotFoundPage';
import RecipeViewPage from './RecipeViewPage';
import PostRecipePage from './PostRecipePage';
import IndexPage from './IndexPage';
import EditRecipePage from './EditRecipePage';
import HomePage from './Homepage';
import CategoryPage from './CategoryPage';

export {
  SigninPage,
  SignupPage,
  DashboardPage,
  NotFoundPage,
  RecipeViewPage,
  PostRecipePage,
  IndexPage,
  EditRecipePage,
  HomePage,
  CategoryPage
};
