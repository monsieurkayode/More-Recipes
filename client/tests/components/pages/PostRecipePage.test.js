/* eslint-disable no-undef */
import React from 'react';

import { PostRecipePage } from '../../../src/components/pages';

const renderSpy = jest.spyOn(PostRecipePage.prototype, 'render');

const setup = () => {
  const props = {
    history: {}
  };

  const wrapper = shallow(<PostRecipePage {...props} />);

  return {
    wrapper
  };
};

describe('The PostRecipePage Component', () => {
  it('should render correctly',
    (done) => {
      const { wrapper } = setup();
      expect(toJson(wrapper)).toMatchSnapshot();
      done();
    });

  it('should call render method',
    (done) => {
      expect(renderSpy).toHaveBeenCalled();
      done();
    });
});
