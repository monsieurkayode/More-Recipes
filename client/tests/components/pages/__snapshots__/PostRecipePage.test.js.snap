// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The PostRecipePage Component should render correctly 1`] = `
<div>
  <Connect(HomeNavbar) />
  <div
    className="row center-align"
  >
    <div
      className="col s12 l6 m8 offset-l3 offset-m2"
    >
      <h5
        className="teal-text"
      >
        Create Recipe
      </h5>
    </div>
  </div>
  <ReduxForm
    history={Object {}}
  />
  <Connect(SideNav) />
</div>
`;
